sendto_silhouette.py

```diff
@@ -555,6 +563,11 @@ def multipassOvercut(self, paths, multipass, reversetoggle, overcut):
 
 
     def effect(self):
+        reg_origin_X = self.options.regoriginx
+        reg_origin_Y = self.options.regoriginy
+        reg_width = self.options.regwidth or self.svg.to_dimensional(self.svg.viewport_width, "mm") - reg_origin_X * 2
+        reg_length = self.options.reglength or self.svg.to_dimensional(self.svg.viewport_height, "mm") - reg_origin_Y * 2
```

Comment:
so to auto detect the regmarks and fix #254 this should do... can you test and give your feedback?
```
        try:
            doc_reg_x = self.svg.unit_to_viewport(self.svg.getElementById('regmark-tl').bounding_box(transform=True).x.minimum, "mm")
            doc_reg_y = self.svg.unit_to_viewport(self.svg.getElementById('regmark-tl').bounding_box(transform=True).y.minimum, "mm")
            doc_reg_width = self.svg.unit_to_viewport(self.svg.getElementById('regmark-tr').bounding_box(transform=True).x.maximum), "mm") - doc_reg_x
            doc_reg_length = self.svg.unit_to_viewport(self.svg.getElementById('regmark-bl').bounding_box(transform=True).y.maximum, "mm") - doc_reg_y
        except Exception:
            doc_reg_x = doc_reg_y = doc_reg_width = doc_reg_length = 0
        reg_origin_X = self.options.regoriginx or doc_reg_x
        reg_origin_Y = self.options.regoriginy or doc_reg_y
        reg_width = self.options.regwidth or doc_reg_width or self.svg.to_dimensional(self.svg.viewport_width, "mm") - reg_origin_X * 2
        reg_length = self.options.reglength or doc_reg_length or self.svg.to_dimensional(self.svg.viewport_height, "mm") - reg_origin_Y * 2
```
... maybe there would be some minor additions useful i.e. add a label in inx to explain 0 means autodetection. and possibly go for 0 as default parameters in py and eventually even in inx parameters... and I hope you can do some tests with your hardware?

Reply:
I like it. Especially since you used `bounding_box().x.minimum` that looks much more elegant than my initial `float(regmark_tr_d.split(" ")[4])` attempt.

I've mostly adopted your approach, but added some extra checks to only autocalc Regmark layer if all the regmarks objects exist as expected. Also cleaned up the constants svg IDs. And added some logging.

I also agree in removing the minimum default checks and added extra notes to INX as well.
Hopefully that's all needed for now.

One other changes that we may want to do, is to print the regmark setting in the preview window so people can double check that the values is the same as their document visually. But that is a separate PR in my opinion if required